types/carbon-components-react/lib/components/Tooltip/Tooltip.d.ts

```diff
@@ -1,22 +1,28 @@
 import * as React from "react";
-import { Direction, EmbeddedIconProps, MenuOffsetData, ReactDivAttr } from "../../../typings/shared";
+import { Direction, EmbeddedIconProps, MenuOffsetData, ReactAttr, ReactDivAttr } from "../../../typings/shared";
 
-type ExcludedAttributes = "aria-labelledby" | "onBlur" | "onContextMenu" | "onFocus" | "onMouseOut" | "onMouseOver" | "role";
+type ExcludedAttributes = "onBlur" | "onChange" | "onContextMenu" | "onFocus" | "onMouseOut" | "onMouseOver" | "role";
 interface InheritedProps extends
     Omit<ReactDivAttr, ExcludedAttributes>,
     EmbeddedIconProps
 { }
 
+type TooltipOnChangeEvent<T extends Element> = React.FocusEvent<T> | React.KeyboardEvent<T> | React.MouseEvent<T>;
 export interface TooltipProps extends InheritedProps {
-    direction?: Direction,
-    iconDescription?: string,
-    iconName?: string,
-    menuOffset?: MenuOffsetData | ((menuBody: HTMLElement, menuDirection: TooltipProps["direction"]) => Required<MenuOffsetData> | undefined)
-    open?: boolean,
-    renderIcon?: React.RefForwardingComponent<any, any>,
-    showIcon?: boolean,
-    triggerClassName?: string,
-    triggerText?: React.ReactNode,
+    defaultOpen?: boolean;
+    direction?: Direction;
+    iconName?: string;
+    menuOffset?:
+        MenuOffsetData
+        | ((menuBody: HTMLElement, menuDirection: TooltipProps['direction']) => Required<MenuOffsetData> | undefined);
```

Comment:
Note: The linter was producing bad code for this. I was getting `menuOffset?: | MenuOffsetData | (parenthesized fn)`

Reply:
Now I'm wondering if it's because of the non-parenthesized return result of the fn....

Reply:
Just checked. Running git hook on
```typescript
    menuOffset?:
        MenuOffsetData
        | ((menuBody: HTMLElement, menuDirection: TooltipProps['direction']) => (Required<MenuOffsetData> | undefined));
```
produced
```typescript
    menuOffset?:
        | MenuOffsetData
        | ((menuBody: HTMLElement, menuDirection: TooltipProps['direction']) => Required<MenuOffsetData> | undefined);
```